File: sdk/sdk-js/src/methods/service.ts
import { ServiceInfo } from '../models';
import { HttpClient } from '../http-client';

/**
 * Service-level operations.
 */
export class ServiceMethods {
    private readonly client: HttpClient;

    constructor(client: HttpClient) {
        this.client = client;
    }

    /**
     * Check if the server is healthy.
     * @returns True if the server is healthy.
     */
    async healthCheck(): Promise<boolean> {
        try {
            return await this.client.head('/');
        } catch {
            return false;
        }
    }

    /**
     * Get service information.
     * @returns Service information including version and uptime.
     */
    async getInfo(): Promise<ServiceInfo> {
        const response = await this.client.get<ServiceInfo>('/');
        if (!response.Data) {
            throw new Error('No data returned from server');
        }
        return response.Data;
    }
}

File: sdk/sdk-python/netledger_sdk/methods/service.py
"""Service-level operations for the NetLedger API."""

from ..http_client import HttpClient
from ..models import ServiceInfo


class ServiceMethods:
    """Service-level operations."""

    def __init__(self, client: HttpClient):
        """
        Initialize service methods.

        Args:
            client: The HTTP client to use.
        """
        self._client = client

    def health_check(self) -> bool:
        """
        Check if the server is healthy.

        Returns:
            True if the server is healthy.
        """
        try:
            return self._client.head('/')
        except Exception:
            return False

    def get_info(self) -> ServiceInfo:
        """
        Get service information.

        Returns:
            Service information including version and uptime.

        Raises:
            NetLedgerConnectionError: If unable to connect to the server.
            NetLedgerApiError: If the server returns an error.
        """
        response = self._client.get('/')
        if not response.data:
            raise ValueError('No data returned from server')
        return ServiceInfo.from_dict(response.data)
